src/dvclive/system_metrics.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from typing import Dict, Union
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+
+from .utils import append_dict
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 30
+
+
+class CPUMetricsCallback:
+    def __init__(
+        self,
+        duration: Union[int, float] = 5,
+        interval: Union[int, float] = 1.0,
+        plot: bool = True,
+    ):
+        self.duration = duration
```

Comment:
do we expect those to be accessed / changed? let's make the protected with `_duration`?

Reply:
Fine by me. As a datascientist, I wouldn't care about these values. I would just like this monitoring feature to take as little CPU usage as possible.